sdk/servicebus/service-bus/test/stress/stress_sessionState.ts

```diff
@@ -0,0 +1,67 @@
+/*
+Test Scenario summary:
+- Creates a single session receiver on a session enabled queue.
+- Autolockrenewal is enabled and set to test duration.
+- Set state to value #1, wait for half test durations (12 hours), get state to retrieve value #1
+- Set state to value #2, wait for other half test durations (12 hours), get state to retrieve value #2
+- Set state to value #3 and get state to retrieve value #3
+
+The test assumes no other process is working with the queues defined in here,
+but the queues must be empty and use default configurations before running the test.
+
+For running this test, connection string of the Service Bus namespace and queue name
+must be supplied.
+*/
+
+import { ServiceBusClient, delay, ReceiveMode } from "../../src";
+
+const connectionString = "";
+const queueName = "";
+
+const testDurationInMilliseconds = 60000 * 20; // 24 hours
```

Comment:
Are we planning to have this run for 24h?

Reply:
Yes, the test has been kickstarted for 24 hours. It passes for 3 min, 5 min and 21 min so far.

Reply:
> In the current set up, `setState` is either getting called at the very beginning or right after `getState` is called i.e it doesnt get called after any kind of delay/idle time.
> 
> I would suggest the below
> 
> * set state
> * wait
> * get state, compare value
> * wait
> * set state
> * get state, compare value

This doesn't seem true.
The second setState() is called after delay of half the test duration, correct?

Reply:
The second `setState()` is called after the first `getState()`
The first `getState()` is called after delay of half the test duration.

There is no delay between the second `setState()` and the first `getState()` 
If the first `getState()` succeeded, then it means that the link is alive and well.
Calling `setState()` right after it, doesnt test the case of "test setState() after delay post an active operation"

Reply:
So we agree that there is a delay between first `setState()` and subsequent `getState()` operation, correct? We are now wanting to induce a delay between same two operations but in reverse order? 
Why would this make a difference, and what exactly is an "active operation"?

(Oops, seems comment thread got misplaced when posting :) )

Reply:
What is the objective of this PR? 
- It is to "Ensure sessionState related get and set APIs work in context of long running operations"

How are we doing this? 
- By introducing idle times before executing the APIs

What did we learn from renewlock tests?
- Receivers kept idle for long time die (non sessions)

What does session lock renewal actually do?
- Keep the session receiver from dying

Another observation we had when working with lock renewal with sessions:
- Any operation on the receiver renews the lock on it on the service side automatically.


Keeping all the above in mind, I would say that this stress test for `setState` and `getState` should be such that we get an opportunity to
- have `getState` as the first operation after waiting for idle time (this we do in this PR)
- have `setState` as the first operation after waiting for idle time  (this we dont in this PR)

Reply:
Ok, since there is delay between two `setState()` and two `getState()` operations, the test seemed sufficient to me. Generally speaking, delay between different operations can end up in endless set of combinations if we go that route.

Also, the point about any operation renewing lock automatically does not seem to be true, as SessionLockLost error is getting thrown when the set autoLockRenewal duration gets exceeded. Is there a PR or link to code/test about this? Executing an operation before the time elapse did not extend the lock and operations at the moment of time elapse failed with error. (Hence, if you see the commit history, I needed to give sufficient buffer time on the lock duration for tests to pass.)

Overall, having this delay between different operations doesn't interfere with other case due to introduction of two delays. Hence, updated tests to factor this in as well.

Reply:
About the introduction of process/link idle time, that would depend on how exactly the operations are executing on service side, correct? Looks like additional inputs, clarity on it from service side might help us get this right.
Agree that this type of delay will help test, rule out other assumptions and points of failure in addition to current ones!

Reply:
> Agree that this type of delay will help test, rule out other assumptions and points of failure in addition to current ones

Thank you. This is exactly what was driving my thoughts as well as the initial test plan in https://github.com/Azure/azure-sdk-for-js/issues/2363#issuecomment-485566025

Reply:
Yes, to summarize 
- Messages being sent and received is not required to work with session state
- Operations on session do not automatically renew lock (because service still throws the asynchronous sessionlocklost error)
- Usage of links and implication of process idle time is not known and current scenarios attempt to try out different theories. More on this can perhaps be continued to look into as part of #2337 as already identified.